### ISSUE TITLE
Setting TooltipVisibility(visible: false) prevents tooltips contributing semantics

### ISSUE THREAD

**Issue by math1man** [flutter/flutter#189062]
### Steps to reproduce

1. Wrap an app in `TooltipVisibility(visible: false)` and include an `IconButton` with a `tooltip`.
2. Run the app and use an accessibility service to navigate to the `IconButton`

### Expected results

The `IconButton` should have an accessibility label based on the value of `tooltip` even when tooltip visibility is disabled. We would like to disable the rendering of tooltips when an accessibility service is enabled, because the tooltip rendering is distracting and doesn't offer any value to an accessibility user. Internal context: b/240567940 and b/531912095.

### Actual results

The `IconButton`'s accessibility label is just "button".

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      builder: (context, child) =>
          TooltipVisibility(visible: false, child: child ?? const SizedBox()),
      home: Scaffold(
        body: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            const Center(child: Text('Hello world')),
            Center(
              child: IconButton(
                icon: const Icon(Icons.add),
                onPressed: () {},
                tooltip: 'Add',
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

<img width="3456" height="2234" alt="Image" src="https://github.com/user-attachments/assets/a3df94ef-48f3-41e4-b704-4095ba225d77" />

</details>

### Logs

<details open><summary>Logs</summary>

N/A, google internal

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

N/A, google internal

</details>

**Comment by math1man** [comment]
The problematic code is https://github.com/flutter/flutter/blob/e8a052d61670f5044ba65bf7369f985cce9f8885/packages/flutter/lib/src/material/tooltip.dart#L539-L559

I think it might be sufficient to add a block like

```dart
if (_visible) {
  // ...
} else {
  effectiveChild = Semantics(
    label: excludeFromSemantics ? null : _tooltipMessage, 
    child: effectiveChild,
  );
}
```

**Comment by mbcorona** [comment]
Hi @math1man , thanks for the detailed report, reporting to the accessibility team for investigation.
